mission_3/routes/products.js

```diff
@@ -0,0 +1,335 @@
+        }
+      });
+
+      // send 201 response status code indicating the query was processed successfully
+      // with the Product record
+      res.status(201).json(product);
+    } catch (err) {
+      console.error('An error has occurred: ', err);
+
+      res.status(500).json({ message: "An error has occurred during processing sql" });
+    }
+  })
+
+  // Inquiry all products
+  .get(async (req, res) => {
+    const { 
+      offset = 0, 
+      limit = 10, 
+      sort = 'recent',
+      search = ''
+    } = req.query;
+
+    let productsSort = '';
+
+    switch (sort) {
+      case 'old':
+        productsSort = 'asc';
+        break;
+
+      case 'recent':
+      default:
+        productsSort = 'desc';
+    }
+
+    try {
+      const products = await prisma.product.findMany({
+        select: {
+          id: true,
+          name: true,
+          price: true,
+          createdAt: true
+        },
+        orderBy: {
+          createdAt: productsSort
+        },
+        where: {
+          OR: [
+            {
+              name: {
+                contains: search,
+                mode: 'insensitive',
+             }
+            },
+            {
+              description: {
+                contains: search,
+                mode: 'insensitive',
+              }
+            }
+          ],
+        },
+        skip: offset * limit,
+        take: limit
+      });
+
+      if (products)
+        res.status(200).json(products);
+      else 
+        res.status(404).json({ message: `Cannot find product with ID ${id}` });
+    } catch (err) {
+      console.error('An error has occurred: ', err.message);
+
+      res.status(500).json({ message: "An error has occurred during processing sql" });
+    }
+  });
+
+
+
+router.route('/:id')
+
+  // Get informations of a certain product
+  .get(async (req, res) => {
+    const { id } = req.params;
+    if (!id)
+      return res.status(400).json({ message: "Invalid parameter 'id'" });
+
+    try {
+      const product = await prisma.product.findUnique({
+        where: {
+          id: parseInt(id)
+        }
+      });
+
+      if (product)
+        res.status(200).json(product);
+      else 
+        res.status(404).json({ message: `Cannot find product with ID ${id}` })
+    } catch (err) {
+      console.error('An error has occurred: ', err.message);
+
+      res.status(500).json({ message: "An error has occurred during processing sql" });
+    }
+  })
+
+  // Modify a product property
+  .patch(async (req, res) => {
+    const { id } = req.params;
+    if (!id)
+      return res.status(400).json({ message: "Invalid parameter 'id'" });
+
+    // Columns in model Product
+    const productCols = [
+      "name",
+      "description",
+      "price",
+      "tags"
+    ];
+
+    const filteredBody = Object.entries(req.body)
+      .filter(e => productCols.includes(e[0]))
+      .reduce((obj, ele) => {
+        obj[ele[0]] = ele[1]
+        return obj;
+      }, {});
```

Comment:
가독성이 좋지 않은 것 같습니다... 주석으로 이 구문이 하는 일을 명시해두거나, body값이 많지 않으니 다음과 같이 작성하는 것이 더 좋을 수 있겠네요.

```suggestion
    const filteredBody = {};
    const { name, description, price, tags } = req.body;

    if (name) filteredBody.name = name;
    if (description) filteredBody.description = description;
    if (price) filteredBody.price = price;
    if (tags) filteredBody.tags = tags;
```

Reply:
물론 입력값 검증은 추가로 작성해주셔야해요!